lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -686,17 +689,32 @@ internal class EvaluatingCompiler(
 
         val uniqueNameMeta = expr.metas.find(UniqueNameMeta.TAG) as? UniqueNameMeta
 
+        val fromSourceNames = currentCompilationContext.fromSourceNames
+
         return when(uniqueNameMeta) {
             null -> {
                 val bindingName = BindingName(id, case.toBindingCase())
                 val evalVariableReference = when (compileOptions.undefinedVariable) {
                     UndefinedVariableBehavior.ERROR   ->
                         thunkEnv(metas) { env ->
-                            env.current[bindingName] ?: throw EvaluationException(
-                                "No such binding: ${bindingName.name}",
-                                ErrorCode.EVALUATOR_BINDING_DOES_NOT_EXIST,
-                                errorContextFrom(metas).also { it[Property.BINDING_NAME] = bindingName.name },
-                                internal = false)
+                            val value = env.current[bindingName]
+                            if(value == null) {
```

Comment:
Is `when` preferred even in the binary case? The general Kotlin [style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#if-versus-when) says "Prefer using `if` for binary conditions instead of `when`. And "Prefer using `when` if there are three or more options."

The PartiQL specific [style guide](https://github.com/partiql/partiql-lang-kotlin/blob/master/docs/dev/CODE-STYLE.md#control-flow) says to "Use `when` instead of `if else if` when possible..."

Reply:
The PartiQL specific style guide should be probably be reviewed and possibly updated.  It was written a while back and should be taken with a grain of salt.

In this case I prefer `when` because it allows inlining the variable declaration inside the `when(...)` which reduces syntactic overhead.  This is however just my opinion which is not any kind of law, of course.

I prefer to use `when` any time the condition is not a boolean value.  i.e. I wouldn't use it for `when(foo == 1) { true -> ... false -> }`.

Reply:
Didn't know about the inlining. Changed to use `when`